### ISSUE TITLE
Bus error (SIGBUS)

### ISSUE THREAD

**Issue by tanya-kta** [physarumAdv/minds_crawl#31]
When running in the debug mode with compilation for GPU, `SIGBUS` is raised in `main.cu` inside `main` while trying to get access to `simulation_map->nodes`

**Comment by kolayne** [comment]
Thanks for your report! Unfortunately, I am not able to reproduce this. I was running the version from `master` (e150e17) with `5` as the argument to `generate_cube()` and the `break` (main.cpp:66 and main.cu:166) commented out because I don't have a visualization running.

When running the project in the debug mode with GPU compilation, I get the `cudaErrorIllegalAddress: an illegal memory access was encountered` error message in my stderr, the program exits with the 188 exit code (if you try to investigate it a bit deeper, you're going to figure out, that the illegal memory access happens inside of the `init_simulation_objects` global function, whose only work now is to create `SimulationMap`, so I think the error's origin is somewhere there. However, I am not 100% sure about it, the error could also be raised because of #32. I am going to dig deeper around it later).
When running in the debug mode with CPU compilation, I get the `Request failed, error: Failed to connect: Connection refused
Error sending http request to visualization. Stopping the simulation process` error in my stderr and the program exits with the 139 exit code (SIGSEGV). By the way, this error seems to exist during the whole project's life...

**Comment by kolayne** [comment]
@tanya-kta, are you still able to reproduce the issue?

**Comment by tanya-kta** [comment]
Yeah, I am still able to reproduce the issue. (May be I'm doing something wrong, but I don't think so). I runned the last for today version from `master` in the debug mode with GPU compilation with `5` as the argument to `generate_cube()` and got `SIGBUS` again, `main.cu`, 125.

> @tanya-kta, are you still able to reproduce the issue?

**Comment by kolayne** [comment]
@tanya-kta, I guess, I see. As far as I'm concerned, the problem, causing the error, is logical. I've opened a new issue (#38) because I am not sure its fix will also remove this issue, but I believe they are describing the same problem. Please, take a look at it

**Comment by kolayne** [comment]
Please, check if you can reproduce the same issue in #40. It is not intended to fix the problem, but it might have fixed it
